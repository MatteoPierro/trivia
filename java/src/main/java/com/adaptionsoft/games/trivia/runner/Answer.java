package com.adaptionsoft.games.trivia.runner;

public class Answer {
    private static final int INCORRECT_ANSWER = 7;
    private final int answer;

    public Answer(int answer) {
        this.answer = answer;
    }

    public boolean isCorrect() {
        return answer != INCORRECT_ANSWER;
    }
}
