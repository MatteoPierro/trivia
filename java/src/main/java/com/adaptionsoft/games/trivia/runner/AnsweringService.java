package com.adaptionsoft.games.trivia.runner;

import java.util.Random;

public class AnsweringService {
    private Random random;

    public AnsweringService(Random random) {
        this.random = random;
    }

    public Answer answer() {
        int answer = random.nextInt(9);
        return new Answer(answer);
    }
 }
