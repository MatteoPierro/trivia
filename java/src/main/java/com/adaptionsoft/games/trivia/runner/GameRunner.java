
package com.adaptionsoft.games.trivia.runner;
import java.util.Random;

import com.adaptionsoft.games.uglytrivia.Game;


public class GameRunner {

	private static boolean notAWinner;

	public static void main(String[] args) {
		Random rand = new Random();
		runGame(rand);
	}

	public static void runGame(Random rand){
		runGame(rand, "Chet", "Pat", "Sue");
	}

	public static void runGame(Random rand, String... players) {
		Game aGame = new Game();

		for (String player:players) {
			aGame.add(player);

		}

		do {

			aGame.roll(rand.nextInt(5) + 1);

			if (rand.nextInt(9) == 7) {
				notAWinner = aGame.wrongAnswer();
			} else {
				notAWinner = aGame.wasCorrectlyAnswered();
			}



		} while (notAWinner);
	}
}
