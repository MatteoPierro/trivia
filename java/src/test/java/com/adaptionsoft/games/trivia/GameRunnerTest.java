package com.adaptionsoft.games.trivia;

import com.adaptionsoft.games.trivia.runner.GameRunner;
import org.approvaltests.Approvals;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.util.Random;

public class GameRunnerTest {

	@Test
	public void run_a_game() throws Exception {
		Integer[] seeds = {1,2};
		Approvals.verifyAll(seeds,seed -> runGame(seed));
	}

	private String runGame(int seed) {
		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
		PrintStream printStream = new PrintStream(outputStream, true);
		PrintStream oldOut = System.out;
		System.setOut(printStream);

		GameRunner.run(new Random(seed));

		System.setOut(oldOut);
		return outputStream.toString();
	}
}
