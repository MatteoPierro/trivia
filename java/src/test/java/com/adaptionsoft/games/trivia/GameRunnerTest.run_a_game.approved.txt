[1, ] => Index 0 out of bounds for length 0 
[1, Chet] => Chet was added
They are player number 1
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 9
The category is Science
Science Question 1
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 2
The category is Sports
Sports Question 0
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 6
The category is Sports
Sports Question 1
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 11
The category is Rock
Rock Question 0
Question was incorrectly answered
Chet was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 2
The category is Sports
Sports Question 2
Answer was correct!!!!
Chet now has 6 Gold Coins.
 
[1, Chet,Pat] => Chet was added
They are player number 1
Pat was added
They are player number 2
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 6
The category is Sports
Sports Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 8
The category is Pop
Pop Question 0
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 10
The category is Sports
Sports Question 1
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 1
The category is Science
Science Question 1
Question was incorrectly answered
Pat was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet's new location is 1
The category is Science
Science Question 2
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 3
Pat is getting out of the penalty box
Pat's new location is 4
The category is Pop
Pop Question 1
Question was incorrectly answered
Pat was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet's new location is 4
The category is Pop
Pop Question 2
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 2
Pat is not getting out of the penalty box
Chet is the current player
They have rolled a 2
Chet's new location is 6
The category is Sports
Sports Question 2
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[1, Chet,Pat,Sue] => Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Sue now has 1 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 6
The category is Sports
Sports Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 7
The category is Rock
Rock Question 1
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 10
The category is Sports
Sports Question 1
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet's new location is 9
The category is Science
Science Question 2
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 10
The category is Sports
Sports Question 2
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 3
Sue is getting out of the penalty box
Sue's new location is 1
The category is Science
Science Question 3
Answer was correct!!!!
Sue now has 2 Gold Coins.
Chet is the current player
They have rolled a 2
Chet's new location is 11
The category is Rock
Rock Question 2
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 2
Pat is not getting out of the penalty box
Sue is the current player
They have rolled a 1
Sue is getting out of the penalty box
Sue's new location is 2
The category is Sports
Sports Question 3
Answer was correct!!!!
Sue now has 3 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 4
Pat is not getting out of the penalty box
Sue is the current player
They have rolled a 3
Sue is getting out of the penalty box
Sue's new location is 5
The category is Science
Science Question 4
Answer was correct!!!!
Sue now has 4 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 5
The category is Science
Science Question 5
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[1, Chet,Pat,Sue,Foo] => Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Foo was added
They are player number 4
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Sue now has 1 Gold Coins.
Foo is the current player
They have rolled a 5
Foo's new location is 5
The category is Science
Science Question 2
Answer was corrent!!!!
Foo now has 1 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 5
The category is Science
Science Question 3
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 8
The category is Pop
Pop Question 0
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 3
Sue's new location is 8
The category is Pop
Pop Question 1
Answer was corrent!!!!
Sue now has 2 Gold Coins.
Foo is the current player
They have rolled a 3
Foo's new location is 8
The category is Pop
Pop Question 2
Question was incorrectly answered
Foo was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet's new location is 8
The category is Pop
Pop Question 3
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 2
Pat is not getting out of the penalty box
Sue is the current player
They have rolled a 2
Sue's new location is 10
The category is Sports
Sports Question 0
Answer was corrent!!!!
Sue now has 3 Gold Coins.
Foo is the current player
They have rolled a 1
Foo is getting out of the penalty box
Foo's new location is 9
The category is Science
Science Question 4
Answer was correct!!!!
Foo now has 2 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 1
The category is Science
Science Question 5
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 4
Pat is not getting out of the penalty box
Sue is the current player
They have rolled a 3
Sue's new location is 1
The category is Science
Science Question 6
Answer was corrent!!!!
Sue now has 4 Gold Coins.
Foo is the current player
They have rolled a 1
Foo is getting out of the penalty box
Foo's new location is 10
The category is Sports
Sports Question 1
Answer was correct!!!!
Foo now has 3 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 2
The category is Sports
Sports Question 2
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 4
Pat is not getting out of the penalty box
Sue is the current player
They have rolled a 5
Sue's new location is 6
The category is Sports
Sports Question 3
Answer was corrent!!!!
Sue now has 5 Gold Coins.
Foo is the current player
They have rolled a 1
Foo is getting out of the penalty box
Foo's new location is 11
The category is Rock
Rock Question 1
Answer was correct!!!!
Foo now has 4 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 3
The category is Rock
Rock Question 2
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[1, Chet,Pat,Sue,Foo,Bar] => Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Foo was added
They are player number 4
Bar was added
They are player number 5
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Sue now has 1 Gold Coins.
Foo is the current player
They have rolled a 5
Foo's new location is 5
The category is Science
Science Question 2
Answer was corrent!!!!
Foo now has 1 Gold Coins.
Bar is the current player
They have rolled a 4
Bar's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Bar now has 1 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 6
The category is Sports
Sports Question 0
Question was incorrectly answered
Chet was sent to the penalty box
Pat is the current player
They have rolled a 3
Pat's new location is 6
The category is Sports
Sports Question 1
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Sue is the current player
They have rolled a 3
Sue's new location is 8
The category is Pop
Pop Question 1
Question was incorrectly answered
Sue was sent to the penalty box
Foo is the current player
They have rolled a 3
Foo's new location is 8
The category is Pop
Pop Question 2
Answer was corrent!!!!
Foo now has 2 Gold Coins.
Bar is the current player
They have rolled a 2
Bar's new location is 6
The category is Sports
Sports Question 2
Answer was corrent!!!!
Bar now has 2 Gold Coins.
Chet is the current player
They have rolled a 2
Chet is not getting out of the penalty box
Pat is the current player
They have rolled a 1
Pat's new location is 7
The category is Rock
Rock Question 1
Answer was corrent!!!!
Pat now has 3 Gold Coins.
Sue is the current player
They have rolled a 5
Sue is getting out of the penalty box
Sue's new location is 1
The category is Science
Science Question 3
Answer was correct!!!!
Sue now has 2 Gold Coins.
Foo is the current player
They have rolled a 4
Foo's new location is 0
The category is Pop
Pop Question 3
Answer was corrent!!!!
Foo now has 3 Gold Coins.
Bar is the current player
They have rolled a 3
Bar's new location is 9
The category is Science
Science Question 4
Answer was corrent!!!!
Bar now has 3 Gold Coins.
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 2
Answer was correct!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 8
The category is Pop
Pop Question 4
Answer was corrent!!!!
Pat now has 4 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Foo is the current player
They have rolled a 5
Foo's new location is 5
The category is Science
Science Question 5
Answer was corrent!!!!
Foo now has 4 Gold Coins.
Bar is the current player
They have rolled a 1
Bar's new location is 10
The category is Sports
Sports Question 3
Answer was corrent!!!!
Bar now has 4 Gold Coins.
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 8
The category is Pop
Pop Question 5
Answer was correct!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 11
The category is Rock
Rock Question 3
Answer was corrent!!!!
Pat now has 5 Gold Coins.
Sue is the current player
They have rolled a 3
Sue is getting out of the penalty box
Sue's new location is 4
The category is Pop
Pop Question 6
Answer was correct!!!!
Sue now has 3 Gold Coins.
Foo is the current player
They have rolled a 1
Foo's new location is 6
The category is Sports
Sports Question 4
Answer was corrent!!!!
Foo now has 5 Gold Coins.
Bar is the current player
They have rolled a 1
Bar's new location is 11
The category is Rock
Rock Question 4
Answer was corrent!!!!
Bar now has 5 Gold Coins.
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 11
The category is Rock
Rock Question 5
Answer was correct!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 2
Pat's new location is 1
The category is Science
Science Question 6
Answer was corrent!!!!
Pat now has 6 Gold Coins.
 
[1, Chet,Pat,Sue,Foo,Bar,Bizz] => Index 6 out of bounds for length 6 
[1, Chet,Pat,Sue,Foo,Bar,Bizz,Dho] => Index 6 out of bounds for length 6 
[2, ] => Index 0 out of bounds for length 0 
[2, Chet] => Chet was added
They are player number 1
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 5
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 10
The category is Sports
Sports Question 0
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Chet is the current player
They have rolled a 2
Chet's new location is 0
The category is Pop
Pop Question 1
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 8
The category is Pop
Pop Question 2
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[2, Chet,Pat] => Chet was added
They are player number 1
Pat was added
They are player number 2
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 9
The category is Science
Science Question 1
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 2
Pat's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 0
The category is Pop
Pop Question 1
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 8
The category is Pop
Pop Question 2
Answer was corrent!!!!
Pat now has 3 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 5
The category is Science
Science Question 2
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 1
The category is Science
Science Question 3
Answer was corrent!!!!
Pat now has 4 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 8
The category is Pop
Pop Question 3
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 2
The category is Sports
Sports Question 0
Answer was corrent!!!!
Pat now has 5 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 0
The category is Pop
Pop Question 4
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[2, Chet,Pat,Sue] => Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Sue now has 1 Gold Coins.
Chet is the current player
They have rolled a 2
Chet's new location is 6
The category is Sports
Sports Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 4
The category is Pop
Pop Question 1
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 10
The category is Sports
Sports Question 1
Answer was corrent!!!!
Sue now has 2 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 11
The category is Rock
Rock Question 0
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 9
The category is Science
Science Question 2
Answer was corrent!!!!
Pat now has 3 Gold Coins.
Sue is the current player
They have rolled a 3
Sue's new location is 1
The category is Science
Science Question 3
Answer was corrent!!!!
Sue now has 3 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 0
The category is Pop
Pop Question 2
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 1
The category is Science
Science Question 4
Answer was corrent!!!!
Pat now has 4 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 6
The category is Sports
Sports Question 2
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 2
Chet's new location is 2
The category is Sports
Sports Question 3
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 4
The category is Pop
Pop Question 3
Answer was corrent!!!!
Pat now has 5 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet's new location is 7
The category is Rock
Rock Question 1
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[2, Chet,Pat,Sue,Foo] => Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Foo was added
They are player number 4
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Sue now has 1 Gold Coins.
Foo is the current player
They have rolled a 2
Foo's new location is 2
The category is Sports
Sports Question 0
Answer was corrent!!!!
Foo now has 1 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 7
The category is Rock
Rock Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 6
The category is Sports
Sports Question 1
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 10
The category is Sports
Sports Question 2
Answer was corrent!!!!
Sue now has 2 Gold Coins.
Foo is the current player
They have rolled a 5
Foo's new location is 7
The category is Rock
Rock Question 1
Answer was corrent!!!!
Foo now has 2 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 10
The category is Sports
Sports Question 3
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 7
The category is Rock
Rock Question 2
Answer was corrent!!!!
Pat now has 3 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 2
The category is Sports
Sports Question 4
Answer was corrent!!!!
Sue now has 3 Gold Coins.
Foo is the current player
They have rolled a 5
Foo's new location is 0
The category is Pop
Pop Question 1
Question was incorrectly answered
Foo was sent to the penalty box
Chet is the current player
They have rolled a 2
Chet's new location is 0
The category is Pop
Pop Question 2
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 3
Pat's new location is 10
The category is Sports
Sports Question 5
Answer was corrent!!!!
Pat now has 4 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 6
The category is Sports
Sports Question 6
Answer was corrent!!!!
Sue now has 4 Gold Coins.
Foo is the current player
They have rolled a 5
Foo is getting out of the penalty box
Foo's new location is 5
The category is Science
Science Question 2
Answer was correct!!!!
Foo now has 3 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 3
The category is Rock
Rock Question 3
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 11
The category is Rock
Rock Question 4
Answer was corrent!!!!
Pat now has 5 Gold Coins.
Sue is the current player
They have rolled a 3
Sue's new location is 9
The category is Science
Science Question 3
Question was incorrectly answered
Sue was sent to the penalty box
Foo is the current player
They have rolled a 3
Foo is getting out of the penalty box
Foo's new location is 8
The category is Pop
Pop Question 3
Answer was correct!!!!
Foo now has 4 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 6
The category is Sports
Sports Question 7
Question was incorrectly answered
Chet was sent to the penalty box
Pat is the current player
They have rolled a 4
Pat's new location is 3
The category is Rock
Rock Question 5
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 3
Sue is getting out of the penalty box
Sue's new location is 0
The category is Pop
Pop Question 4
Answer was correct!!!!
Sue now has 5 Gold Coins.
Foo is the current player
They have rolled a 5
Foo is getting out of the penalty box
Foo's new location is 1
The category is Science
Science Question 4
Answer was correct!!!!
Foo now has 5 Gold Coins.
Chet is the current player
They have rolled a 2
Chet is not getting out of the penalty box
Pat is the current player
They have rolled a 5
Pat is getting out of the penalty box
Pat's new location is 8
The category is Pop
Pop Question 5
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Question was incorrectly answered
Sue was sent to the penalty box
Foo is the current player
They have rolled a 5
Foo is getting out of the penalty box
Foo's new location is 6
The category is Sports
Sports Question 8
Question was incorrectly answered
Foo was sent to the penalty box
Chet is the current player
They have rolled a 4
Chet is not getting out of the penalty box
Pat is the current player
They have rolled a 1
Pat is getting out of the penalty box
Pat's new location is 9
The category is Science
Science Question 5
Answer was correct!!!!
Pat now has 6 Gold Coins.
 
[2, Chet,Pat,Sue,Foo,Bar] => Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Foo was added
They are player number 4
Bar was added
They are player number 5
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 1
Pat's new location is 1
The category is Science
Science Question 0
Answer was corrent!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Sue now has 1 Gold Coins.
Foo is the current player
They have rolled a 2
Foo's new location is 2
The category is Sports
Sports Question 0
Answer was corrent!!!!
Foo now has 1 Gold Coins.
Bar is the current player
They have rolled a 3
Bar's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Bar now has 1 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 9
The category is Science
Science Question 2
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 6
The category is Sports
Sports Question 1
Answer was corrent!!!!
Pat now has 2 Gold Coins.
Sue is the current player
They have rolled a 5
Sue's new location is 10
The category is Sports
Sports Question 2
Answer was corrent!!!!
Sue now has 2 Gold Coins.
Foo is the current player
They have rolled a 3
Foo's new location is 5
The category is Science
Science Question 3
Answer was corrent!!!!
Foo now has 2 Gold Coins.
Bar is the current player
They have rolled a 1
Bar's new location is 4
The category is Pop
Pop Question 1
Answer was corrent!!!!
Bar now has 2 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 1
The category is Science
Science Question 4
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 5
Pat's new location is 11
The category is Rock
Rock Question 1
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 2
Sue's new location is 0
The category is Pop
Pop Question 2
Answer was corrent!!!!
Sue now has 3 Gold Coins.
Foo is the current player
They have rolled a 3
Foo's new location is 8
The category is Pop
Pop Question 3
Answer was corrent!!!!
Foo now has 3 Gold Coins.
Bar is the current player
They have rolled a 4
Bar's new location is 8
The category is Pop
Pop Question 4
Answer was corrent!!!!
Bar now has 3 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 6
The category is Sports
Sports Question 3
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 3
Pat is getting out of the penalty box
Pat's new location is 2
The category is Sports
Sports Question 4
Answer was correct!!!!
Pat now has 3 Gold Coins.
Sue is the current player
They have rolled a 1
Sue's new location is 1
The category is Science
Science Question 5
Answer was corrent!!!!
Sue now has 4 Gold Coins.
Foo is the current player
They have rolled a 3
Foo's new location is 11
The category is Rock
Rock Question 2
Question was incorrectly answered
Foo was sent to the penalty box
Bar is the current player
They have rolled a 3
Bar's new location is 11
The category is Rock
Rock Question 3
Answer was corrent!!!!
Bar now has 4 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 9
The category is Science
Science Question 6
Question was incorrectly answered
Chet was sent to the penalty box
Pat is the current player
They have rolled a 4
Pat is not getting out of the penalty box
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 3
Sue's new location is 4
The category is Pop
Pop Question 5
Answer was corrent!!!!
Sue now has 5 Gold Coins.
Foo is the current player
They have rolled a 5
Foo is getting out of the penalty box
Foo's new location is 4
The category is Pop
Pop Question 6
Answer was correct!!!!
Foo now has 4 Gold Coins.
Bar is the current player
They have rolled a 2
Bar's new location is 1
The category is Science
Science Question 7
Answer was corrent!!!!
Bar now has 5 Gold Coins.
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 2
The category is Sports
Sports Question 5
Question was incorrectly answered
Chet was sent to the penalty box
Pat is the current player
They have rolled a 4
Pat is not getting out of the penalty box
Question was incorrectly answered
Pat was sent to the penalty box
Sue is the current player
They have rolled a 5
Sue's new location is 9
The category is Science
Science Question 8
Question was incorrectly answered
Sue was sent to the penalty box
Foo is the current player
They have rolled a 4
Foo is not getting out of the penalty box
Bar is the current player
They have rolled a 1
Bar's new location is 2
The category is Sports
Sports Question 6
Answer was corrent!!!!
Bar now has 6 Gold Coins.
 
[2, Chet,Pat,Sue,Foo,Bar,Bizz] => Index 6 out of bounds for length 6 
[2, Chet,Pat,Sue,Foo,Bar,Bizz,Dho] => Index 6 out of bounds for length 6 
